internal/server/config/config.go

```diff
@@ -333,9 +349,35 @@ func (cfg *Config) Verify() error {
 		}
 	}
 
+	if cfg.RequestTimeout < 0 {
+		return errors.New("request duration must be non-negative time duration")
+	}
+
+	if cfg.RequestTimeout == 0 && cfg.HTTP.Enabled && cfg.HTTP.UpstreamTimeout < 0 {
+		return errors.New("http upstream timeout must be non-negative time duration")
```

Comment:
Sorry, I didn't notice this previously, but in the error messages above we use the actual config value in the error message. I suggest we do the same here and below and strive to remain consistent.

> http.upstreamTimeout must be a non-negative time duration